### ISSUE TITLE
Lagging as ****

### ISSUE THREAD

**Issue by ServOKio** [rodydavis/infinite_canvas#7]
On the desktop, I have huge problems with the fact that the workspace:
1. It lags a lot when moving
2. When zooming in
3. When dragging
and so on

**Comment by rodydavis** [comment]
Can you create a reproducible example?

Just tested last night for something and the. Demo is still smooth for me.

**Comment by rodydavis** [comment]
Also how many nodes/edges?

And please share a video if possible

**Comment by ServOKio** [comment]
You can open your own example

It all depends on the size of the workspace. Yes, it works well on small screens, starts lagging on 2560x1440 monitors and slows down even more on 4k monitors

https://github.com/user-attachments/assets/dd931600-f3eb-4501-8d5c-675f87a25a5d

**Comment by rodydavis** [comment]
Thanks for sharing this!

I will see why DPI is playing such a part. I think it might be the grid background.

I'll do some tests to be sure

**Comment by jtkeyva** [comment]
yes it's lagging as well for me quite a bit on your demo. any updates?

**Comment by ServOKio** [comment]
2025 is here, hello ?

**Comment by rodydavis** [comment]
Happy to review PRs!

**Comment by abdullahmatani** [comment]
> Thanks for sharing this!
> 
> I will see why DPI is playing such a part. I think it might be the grid background.
> 
> I'll do some tests to be sure

I think it is the background. I have built a bowtie drawing tool using package, and when setting the default backgroundBuilder in InfiniteCanvas, there is some lag. When replacing the default with a background builder that returns an empty Container or SizedBox, the lag is reduced significantly.

**Comment by abdullahmatani** [comment]
> Thanks for sharing this!
> 
> I will see why DPI is playing such a part. I think it might be the grid background.
> 
> I'll do some tests to be sure

I think it is the background. I have built a bowtie drawing tool using package, and when setting the default backgroundBuilder in InfiniteCanvas, there is some lag. When replacing the default with a background builder that returns an empty Container or SizedBox, the lag is reduced significantly.

**Comment by EnziinSystem** [comment]
I also ran the example on Windows at 1920*1080, it was very slow in all 2D operations.
It is unbelievable that a platform like Flutter, with a Skia renderer behind it, is inferior to a 19th-century renderer, Cairo 2D.

**Comment by EnziinSystem** [comment]
Update:

You need to run the app in release mode instead of debug mode, then the 2D performance issues will disappear.

It's not because of the window size, the reason is that in debug mode, Dart VM will run and support hot reload, but in release mode everything will be native and the app will run smoothly.

Build in release mode:
```
flutter build windows
```
And run the app directly in the folder:
```
example\build\windows\x64\runner\Release\example.exe
```

**Comment by jtkeyva** [comment]
yeah the current online demo works smoothly https://rodydavis.github.io/infinite_canvas/
